### PR TITLE
Transform single-piece gallery expandos

### LINKED ISSUE #3227: Treat single-image galleries/albums as single images
@larsjohnsen, does one of your show image refactors already handle this?

### PR DESCRIPTION
**Pull request by larsjohnsen**
Resolves #3227

### CODE REVIEW

**Comment by andytuba** [comment]
Should  that first commit go on a separate pr?

File: lib/modules/showImages.js [hunk 1]
```diff
@@ -830,10 +830,21 @@ const retrieveExpandoOptions = _.memoize(async (thing, { siteModule, detectResul
 		thing.isPost() && !thing.isSelfPost() &&
 		siteModule.domains.length && siteModule.attribution !== false;
 
+	const isMuted = type => ['IMAGE', 'TEXT'].includes(type);
+
+	if (mediaOptions.type === 'GALLERY') {
+		mediaOptions.muted = isMuted(mediaOptions.src[0].type);
```
Comment (erikdesjardins) [inline lib/modules/showImages.js]: Maybe `mediaOptions.muted = mediaOptions.src.every(s => isMuted(s.type));`

File: lib/modules/showImages.js [hunk 2]
```diff
@@ -830,10 +830,10 @@ const retrieveExpandoOptions = _.memoize(async (thing, { siteModule, detectResul
 		thing.isPost() && !thing.isSelfPost() &&
 		siteModule.domains.length && siteModule.attribution !== false;
 
-	const isMuted = type => ['IMAGE', 'TEXT'].includes(type);
+	const isMuted = ({ muted, type }) => muted || ['IMAGE', 'TEXT'].includes(type);
 
 	if (mediaOptions.type === 'GALLERY') {
-		mediaOptions.muted = isMuted(mediaOptions.src[0].type);
+		mediaOptions.muted = _.every(mediaOptions.src, isMuted);
```
Comment (erikdesjardins) [inline lib/modules/showImages.js]: Why lodash instead of `Array.prototype.every`? (I think this will always be an array?)

Comment (larsjohnsen) [inline lib/modules/showImages.js]: Brain fart. Fixed now.

**Pull request merged**